src/components/button/button.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import IVariant from "../../interfaces/IVariant";
+import '../../styles/components/_button.scss';
+import classNamesDefault from '../../utils/class-names-default';
+
+class Button extends React.Component<IProps, any> {
+    public render() {
+        return (
+            <button
+                className={classNamesDefault(this.props)}
+                disabled={this.props.disabled}
+                onClick={this.a}
```

Comment:
a diye bir function ismi kullanmayalim

Reply:
onClick prop olarak alinip callback olarak cagirilmali

Reply:
Denemek için yazılmıştı function silindi ve onClick artık prop olarak cagiliyor, düzeltildi.